x-pack/plugins/fleet/server/tasks/check_deleted_files_task.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { throwUnrecoverableError } from '@kbn/task-manager-plugin/server';
+import type { LoggerFactory } from '@kbn/core/server';
+
+import {
+  fileIdsWithoutChunksByIndex,
+  getFilesByStatus,
+  updateFilesStatus,
+} from '../services/files';
+
+export const TYPE = 'fleet:check-deleted-files-task';
+export const VERSION = '1.0.0';
+const TITLE = 'Fleet Deleted Files Periodic Tasks';
+const TIMEOUT = '2m';
+const SCOPE = ['fleet'];
+const INTERVAL = '1d';
+
+interface CheckDeletedFilesTaskSetupContract {
+  core: CoreSetup;
+  taskManager: TaskManagerSetupContract;
+  logFactory: LoggerFactory;
+}
+
+interface CheckDeletedFilesTaskStartContract {
+  taskManager: TaskManagerStartContract;
+}
+
+export class CheckDeletedFilesTask {
+  private logger: Logger;
+  private wasStarted: boolean = false;
+
+  constructor(setupContract: CheckDeletedFilesTaskSetupContract) {
+    const { core, taskManager, logFactory } = setupContract;
+    this.logger = logFactory.get(this.taskId);
+
+    taskManager.registerTaskDefinitions({
+      [TYPE]: {
+        title: TITLE,
+        timeout: TIMEOUT,
```

Comment:
I see you're setting a timeout of `2m`. When a task hits that 2 minute timeout, task manager will try to cancel the best it can it by calling the registered cancel function, which for this task is empty. This means the task will continue to run to completion, no matter how long it ends up taking. Would it make sense to add cancellation behavior that tries to short circuit execution when the timeout is reached? You could maybe pass an abort signal to the esClient.

Reply:
that makes sense, I'll make some changes for this. thanks for the tip.